### PR TITLE
axios migration refactoring useFetchClient and getFetchclient

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

We want to move the implementation of the getFetchClient and useFetchClient from the admin to the helper-plugin to share it

### Why is it needed?

Because otherwise, they are not reachable from other plugins

### How to test it?

in the Dashboard one API is already called with the get method from the _useFetchClient_ defined in the helper-plugin, it is the `/admin/telemetry-properties` call

### CODE REVIEW

**Review by ronronscelestes — COMMENTED** [review]
it overall looks good to me, thank you for working on it!
small questions do some docs need to be updated? (importing from hp but also the `.env` not used by default)

**Comment by simotae14** [comment]
good catch @ronronscelestes

**Comment by simotae14** [comment]
Hello @gu-stav and @joshuaellis please let me know if this solution work for you, based also on your request in this other PR https://github.com/strapi/strapi/pull/15350. I know it is not the best solution, but it can be a good first attempt to unlock other related tickets that I have in the Backlog (https://strapi-inc.atlassian.net/browse/DX-562, https://strapi-inc.atlassian.net/browse/DX-564, https://strapi-inc.atlassian.net/browse/DX-565, https://strapi-inc.atlassian.net/browse/DX-566, https://strapi-inc.atlassian.net/browse/DX-567, https://strapi-inc.atlassian.net/browse/DX-568, https://strapi-inc.atlassian.net/browse/DX-569, https://strapi-inc.atlassian.net/browse/DX-570). After all of this work, we can do another round of the solution trying to pass the STRAPI_ADMIN_BACKEND_URL env var to the helper-plugin. Thanks.

**Review by christiancp100 — APPROVED** [review]
LGTM 🌻

File: packages/core/helper-plugin/lib/src/utils/fetchClient/index.js [hunk 1]
```diff
@@ -36,10 +34,14 @@ export const addInterceptors = (instance) => {
...
+      'Content-Type': 'application/json',
+    },
   });
   addInterceptors(instance);
 
   return instance;
 };
 
-export default fetchClient({ baseURL: process.env.STRAPI_ADMIN_BACKEND_URL });
+export default fetchClient({});
```
Comment (gu-stav) [inline packages/core/helper-plugin/lib/src/utils/fetchClient/index.js:47]: There is no way to pass `baseURL` anymore. Can we remove the empty object in line 47 and the attributes in line 34?

Comment (simotae14) [reply packages/core/helper-plugin/lib/src/utils/fetchClient/index.js:47]: ok

Comment (simotae14) [reply packages/core/helper-plugin/lib/src/utils/fetchClient/index.js]: done

**Review by gu-stav — APPROVED** [review]

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ListView/tests/index.test.js [hunk 2]
```diff
@@ -878,8 +878,8 @@ describe('ADMIN | Pages | API TOKENS | ListPage', () => {
                           class="c5 c31"
                         >
                           <time
-                            datetime="2021-11-15T00:00:00.000Z"
-                            title="11/15/2021 12:00 AM"
```
Comment (gu-stav) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ListView/tests/index.test.js:882]: @simotae14 Do you understand why these have changed? I don't think we should commit this if there is no good reason and I believe it needs to be understood.

Comment (simotae14) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ListView/tests/index.test.js:882]: they are the same but they failed, not just this one, other snapshots tests, I will revert back the code after this attempt

**Review by joshuaellis — APPROVED** [review]
Can we add a link in the `helper-plugin` readme to direct users to the contributor docs if its not there already?

File: docs/docs/core/hooks/use-fetch-client.mdx [hunk 3]
```diff
@@ -57,7 +57,7 @@ The following information is the internal additions we've added to the axios ins
 
 ### Base URL
 
-The default URL will be the one defined in the environment variable: `STRAPI_ADMIN_BACKEND_URL`.
+The default URL will be the one defined in the _getFetchClient_ utility
```
Comment (StrapiShaun) [inline docs/docs/core/hooks/use-fetch-client.mdx:60]:
```suggestion
The default URL is the one defined in the _getFetchClient_ utility
```

**Review by StrapiShaun — APPROVED** [review]
Just one small suggestion to improve the wording. Otherwise LGTM

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**